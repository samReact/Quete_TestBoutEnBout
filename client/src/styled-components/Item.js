import styled from 'styled-components';

export const Name = styled.div`
  text-align: center;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 18px;
  font-weight: bold;
`;

export const Price = styled.div`
  text-align: center;
  font-family: Georgia, 'Times New Roman', Times, serif;
  font-size: 14px;
  font-style: italic;
`;
