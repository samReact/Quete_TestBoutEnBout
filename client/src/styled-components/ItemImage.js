import styled from 'styled-components';

export const Container = styled.div`
  position: relative;
  width: 100%;
  padding-top: 100%;
`;

export const Centering = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-content: center;
`;

export const Image = styled.img`
  max-width: 100%;
  max-height: 100%;
  object-fit: contain;
  margin-bottom: 48px;
`;
