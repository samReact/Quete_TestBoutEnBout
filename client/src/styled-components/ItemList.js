import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const List = styled.ul`
  margin: 128px auto;
  display: grid;
  width: 1024px;
  grid-template-columns: 256px 256px 256px;
  column-gap: 128px;
  row-gap: 128px;
  padding-inline-start: 0;
`;

export const ListElement = styled.li`
  display: block;
`;

export const LinkToDetails = styled(Link)`
  color: initial;
  text-decoration: initial;
`;
