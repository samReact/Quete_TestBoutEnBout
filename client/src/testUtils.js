// eslint-disable-next-line import/prefer-default-export
export const items = [
  {
    id: '1234',
    shortName: 'Glouglou',
    inflatedPrice: 100,
    largestImageUrl: 'http://no.big/deal',
  },
  {
    id: '5678',
    shortName: 'mdr',
    largestImageUrl: 'http://okfine.com/lol',
  },
];
