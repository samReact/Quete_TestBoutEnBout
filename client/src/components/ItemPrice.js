import React from 'react';
import PropTypes from 'prop-types';
import { Price } from '../styled-components/Item';

const ItemPrice = ({ value, render }) =>
  render(value ? `â‚¬${value}` : 'Contact us');

ItemPrice.propTypes = {
  value: PropTypes.number,
  render: PropTypes.func,
};

ItemPrice.defaultProps = {
  value: null,
  render: value => <Price>{value}</Price>,
};

export default ItemPrice;
