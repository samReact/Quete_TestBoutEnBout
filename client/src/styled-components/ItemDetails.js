import styled from 'styled-components';
import { Name as ItemName, Price as ItemPrice } from './Item';

export const Grid = styled.div`
  margin: 128px auto;
  display: grid;
  width: 1024px;
  grid-template-columns: 464px 464px;
  column-gap: 96px;
  padding-inline-start: 0;
`;

export const Name = styled(ItemName)`
  margin-top: 48px;
  text-align: initial;
  font-size: 36px;
`;

export const Price = styled(ItemPrice)`
  margin-top: 16px;
  text-align: initial;
  font-size: 24px;
`;
