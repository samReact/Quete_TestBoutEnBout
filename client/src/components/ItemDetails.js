import React from 'react';
import PropTypes from 'prop-types';
import ItemImage from './ItemImage';
import ItemPrice from './ItemPrice';
import { Grid, Name, Price } from '../styled-components/ItemDetails';

const ItemDetails = ({ shortName, inflatedPrice, imageUrl }) => (
  <Grid>
    <ItemImage imageUrl={imageUrl} />
    <div>
      <Name>{shortName.toUpperCase()}</Name>
      <ItemPrice
        value={inflatedPrice}
        render={value => <Price>{value}</Price>}
      />
    </div>
  </Grid>
);

ItemDetails.propTypes = {
  shortName: PropTypes.string,
  inflatedPrice: PropTypes.number,
  imageUrl: PropTypes.string,
};

ItemDetails.defaultProps = {
  shortName: '',
  inflatedPrice: null,
  imageUrl: '',
};

export default ItemDetails;
