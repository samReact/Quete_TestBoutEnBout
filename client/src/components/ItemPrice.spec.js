import React from 'react';
import { shallow } from 'enzyme';
import { items } from '../testUtils';
import { Price } from '../styled-components/ItemDetails';
import ItemPrice from './ItemPrice';

describe('ItemPrice', () => {
  describe('when passed no value', () => {
    it('renders text "Contact us"', () => {
      const wrapper = shallow(<ItemPrice />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when passed value', () => {
    const value = items[0].inflatedPrice;

    it('renders price value', () => {
      const wrapper = shallow(<ItemPrice value={value} />);
      expect(wrapper).toMatchSnapshot();
    });

    describe('when passed render prop', () => {
      const render = val => <Price>{val}</Price>;

      it('renders price value with render prop', () => {
        const wrapper = shallow(<ItemPrice value={value} render={render} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(Price)).toHaveLength(1);
      });
    });
  });
});
