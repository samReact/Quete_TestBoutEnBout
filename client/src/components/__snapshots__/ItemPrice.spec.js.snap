// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPrice when passed no value renders text "Contact us" 1`] = `
<styled.div>
  Contact us
</styled.div>
`;

exports[`ItemPrice when passed value renders price value 1`] = `
<styled.div>
  €100
</styled.div>
`;

exports[`ItemPrice when passed value when passed render prop renders price value with render prop 1`] = `
<Styled(styled.div)>
  €100
</Styled(styled.div)>
`;
